src/start_server.ts

```diff
@@ -151,6 +156,32 @@ export function getApp(options: ServerOptions): express.Express {
 
   app.use('/components/', polyserve);
 
+  if (options.proxy) {
+    if (options.proxy.path === 'components') {
+      console.error('proxy path can not start with components.');
+      return;
+    }
+
+    const proxyPath = options.proxy.path
+                             .replace('*', '\*')
+                             .replace('?', '\?')
+                             .replace('+', '\+');
+
+    const apiProxy = httpProxy.createProxyServer();
+    const proxyHandler = (req: express.Request, res: express.Response) => {
+      if (req.url.substring(0, proxyPath.length) === proxyPath) {
+        // Remove proxied part from the url
+        req.url = req.url.slice(proxyPath.length + 1);
+      }
+      apiProxy.web(req, res, { target: options.proxy.target }, (error: proxy.ProxyError) => {
+        res.statusCode = error.status || 500;
+        res.end(error.message);
+      });
+    };
+
+    app.all(`/${proxyPath}/*`, proxyHandler);
```

Comment:
Last bit of validation / conditional logic on the path: Handle cases where path starts or ends with `/`.